docs: add how-to for using Ubuntu One authentication (#384)
canonical/craft-store: docs/how-to/login-ubuntu-one-keyring.rst

Author: https://canonical-ubuntu-documentation-library--384.com.readthedocs.build/craft-store/384/how-to/login-ubuntu-one-keyring/

.. _howto-login_ubuntu_one_keyring:

Log in to Charmhub with Ubuntu One
==================================

This guide shows you how to log in to Charmhub using your Ubuntu One credentials and
how to save those credentials securely in your system keyring. This allows you to
retrieve them in future sessions without re-entering your password.

Prerequisites
-------------

Before you begin, ensure you have:

*   An Ubuntu One account.
*   The ``craft-store`` library installed.

> *   A functional system keyring, such as GNOME Keyring or KWallet.

Reviewer: ```suggestion
* An Ubuntu One account.
* The ``craft-store`` library installed.
* A functional system keyring, such as GNOME Keyring or KWallet.
```

Authenticate and save credentials
----------------------------------

> Use :meth:`~craft_store.login.UbuntuOneLogin.login_with` to authenticate. This method

Reviewer: ```suggestion
To authenticate, use the :meth:`~craft_store.login.UbuntuOneLogin.login_with` method. This method
```
In a how-to like this where someone might not follow every step, I find it's nice to state the outcome before the instruction. That gives users a chance to go "Hey, I don't need to do this part."

       permissions=["account-view-packages", "account-register-package"],
   )

Replace ``<email>`` and ``<password>`` with your actual credentials.

Authenticate with OTP
~~~~~~~~~~~~~~~~~~~~~

If your account has two-factor authentication enabled, the first call will raise
:exc:`~craft_store.login.UbuntuOneOtpRequiredError`. Retry with your one-time
password using the ``otp`` argument:

.. code-block:: python

   from craft_store.login import UbuntuOneLogin, UbuntuOneOtpRequiredError

   try:
       UbuntuOneLogin.login_with(
           email="<email>",
           password="<password>",
           base_url="https://api.charmhub.io",
           permissions=["account-view-packages", "account-register-package"],
       )
   except UbuntuOneOtpRequiredError:
       UbuntuOneLogin.login_with(
           email="<email>",
           password="<password>",
           base_url="https://api.charmhub.io",
           permissions=["account-view-packages", "account-register-package"],
           otp="<otp>",
       )

Use the credentials with a store client
---------------------------------------

Use :meth:`~craft_store.publisher.PublisherGateway.with_ubuntu_one` to create a

> gateway that reads your saved credentials from the keyring automatically.

Reviewer: ```suggestion
To create a gateway that reads your saved credentials from the keyring, use the
:meth:`~craft_store.publisher.PublisherGateway.with_ubuntu_one` method.
```
This is just a pet peeve of mine, but I find the word "automatically" is rarely needed in software docs. The purpose of most software is to automate things / perform tasks programmatically.

.. code-block:: python

   from craft_store.publisher import PublisherGateway

   gateway = PublisherGateway.with_ubuntu_one(
       base_url="https://api.charmhub.io",
       namespace="charm",
   )

   # List your registered charms
   for charm in gateway.list_registered_names(include_collaborations=True):
       print(f"{charm.name} [{charm.status}]")

Verify your login
-----------------

> You can use the ``whoami()`` method to verify that you're logged in and to retrieve

Reviewer: ```suggestion
Use the ``whoami()`` method to verify that you're logged in and to retrieve
```
Unnecessary passive voice

information about your account.

.. code-block:: python

   user_info = gateway.whoami()
   print(f"Logged in as: {user_info['account']['display-name']}")

Handle login errors
-------------------

> The ``login_with`` method can raise specific errors if the authentication fails.

Reviewer: ```suggestion
The ``login_with`` method raises specific errors if the authentication fails.
```
I can catch them, but they're always raised, right?
